pkg/controller/sidecarset/sidecarset_pod_event_handler.go

```diff
@@ -57,6 +74,9 @@ func (p *enqueueRequestForPod) addPod(q workqueue.RateLimitingInterface, obj run
 	}
 
 	for _, sidecarSet := range sidecarSets {
+		if pod.DeletionTimestamp != nil {
```

Comment:
I think it is not necessary to execute the expectation function when add pod.

Reply:
on a restart of the controller manager, it's possible a new pod shows up in a state that is already pending deletion. Prevent the pod from being a creation observation.